### PR TITLE
Add rule_id to notifications and move logging statement to universal method.

### PR DESCRIPTION
**Pull request by JTCunning**
@mattrobenolt: Since this is still a little convoluted, let's have `rule_ids` come along with `notification.dispatched` until I get a good sample size of production data.

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4112)

### CODE REVIEW

File: src/sentry/plugins/bases/notify.py [hunk 1]
```diff
@@ -96,6 +92,11 @@ def rule_notify(self, event, futures):
                 rules=rules,
             )
             self.notify(notification)
+            self.logger.info('notification.dispatched', extra={
+                'event_id': event.id,
+                'plugin': self.slug,
+                'rule_id': rules[0] if any(rules) else None,
```
Comment (mattrobenolt) [inline src/sentry/plugins/bases/notify.py]: You don't need `any` here.

File: src/sentry/plugins/bases/notify.py [hunk 2]
```diff
@@ -96,6 +92,11 @@ def rule_notify(self, event, futures):
                 rules=rules,
             )
             self.notify(notification)
+            self.logger.info('notification.dispatched', extra={
```
Comment (mattrobenolt) [inline src/sentry/plugins/bases/notify.py]: Shouldn't this come before `notify()` ? If `notify()` were to raise, this would never be logged.

Comment (mattrobenolt) [inline src/sentry/plugins/bases/notify.py]: Or I guess, maybe that's intentional.

Comment (JTCunning) [inline src/sentry/plugins/bases/notify.py]: Yea.. technically if it goes to raise it was never dispatched.

**Comment by mattrobenolt** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/4112)*

**Pull request merged**